src/lib/userStorage/UserStorageClass.js

```diff
@@ -896,6 +908,25 @@ export class UserStorage {
 
     const fromGDUbi = this.wallet.getUBIAddresses().includes((data.address || '').toLowerCase()) && 'GoodDollar UBI'
 
+    let fromAddy = 'Unknown'
+    let fullFromAddress = 'Unknown'
+    if (get(receiptEvent, 'from')) {
+      fullFromAddress = get(receiptEvent, 'from')
+
+      //do this for printing
+      fromAddy =
+        fullFromAddress.slice(0, 7) + '...' + fullFromAddress.slice(fullFromAddress.length - 5, fullFromAddress.length)
```

Comment:
1.  we have util to shorten string such way
2. we do not display counterpart wallet address, just printing 'Unknown' for G$ txs
3. if this was a requirement see how it's done here https://github.com/GoodDollar/GoodDAPP/pull/4015/files (this is PR for support native txs and there we're showing counterpart address for native txs if there're no username associated)

Reply:
This isn't a full requirement for us - if needed we can remove this portion to leave it saying Unknown.

Reply:
I think you should use the wallet address in case of unknown.
though I believe we already got a utility to shorten a text called `truncateMiddle`

Reply:
> 

@sirpy for this - it would change what Transactions look like in your wallet UI - Lewis and John both told us to leave "Unknown" displayed for a non-GD wallet in the GD wallet.  Can you confirm this is what you want?  

Our WalletChat code is using a different value - so the chat with owner works fine on our end.

Reply:
I dont want to change at the moment how it looks on our feed.
I thought this was for what is displayed in your widget.

Reply:
> I think you should use the wallet address in case of unknown.
> though I believe we already got a utility to shorten a text called `truncateMiddle`

yes I used it in my ethereum PR